nestkernel/modelrange_manager.cpp

```diff
@@ -104,7 +104,7 @@ ModelRangeManager::get_model_id( size_t node_id ) const
       range_idx -= ( range_idx - left ) / 2;
     }
     assert( left + 1 < right );
-    assert( range_idx < modelranges_.size() );
+    assert( static_cast< size_t >( range_idx ) < modelranges_.size() );
```

Comment:
Just to be entirely on the safe side that we do not cast away negative values (which we did not control for in the old version):
```suggestion
    assert( range_idx >= 0 );
    assert( static_cast< size_t >( range_idx ) < modelranges_.size() );
```